simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,787 @@
+    #        print('line:'+str(line))
+            values = line.split()
+            if len(values):
+                K_points = (values)
+                print('k points:'+str(K_points))
+                SP[CUBA.EQUATION_OF_STATE_COEFFICIENT] = K_points
+
+            line = f.next()
+        return line
+
+
+    #I am not sure whether to use datacontainer or particlecontainer - maybe PC goes in DC?
+    def process_atomic_positions(self,f,pc,SP,units='(angstrom)'):
+        print('processing atomic_positions section')
+        try:
+            line = f.next()
+            SP[CUBA.KINEMATIC_VISCOSITY] = units
+            particle_list = []
+            while _ReadState.get_state(_ReadState.ATOMIC_POSITIONS,line) == _ReadState.ATOMIC_POSITIONS:
+                logging.debug('line in atomic positions section:'+str(line))
+                values = line.split()
+                print('values:'+str(values))
+                atom_pos = [0,0,0]
+                i = 0
+                if values[0] in atomtypes:
+                    atomtype = values[0]
+                    atom_id = i
+                    atom_pos[0] = float(values[1]) * 1e-10  #store position in meters; original in Angstrom
+                    atom_pos[1] = float(values[2]) * 1e-10
+                    atom_pos[2] = float(values[3]) * 1e-10
+                    s = str(i)
+                    #make uid using string of index
+                    #uidstring =
+                    #uid = uuid.UUID(uidstring)
+
+                    p = Particle([atom_pos[0],atom_pos[1],atom_pos[2]])  #,uuid.UUID(int=i)
+                    print('uid:'+str(p.uid))
+                    p.data[CUBA.CHEMICAL_SPECIE] = atomtype
+                    particle_list.append(p)
+                    try:
+                        line = f.next()
+                    except StopIteration:
+                        print('EOF')
+                        break
+            if len(particle_list):
+                pc.add_particles(particle_list)
+                  #  part_container.add_particle(p)
+                n=self.count_particles()
+                print('n_particles='+str(n))
+        except StopIteration:
+            return('EOF')
+
+        return pc
+
+    def count_particles(self):
+        n=0
+        for particle in self.pc.iter_particles():
+            n+=1
+        return n
+
+    def start_qe(self,name_in,name_out,path_to_espresso='/usr/bin/pw.x',mpi=False,mpi_Nprocessors=2):
+#        name_in = './test_pw.in'
+#        name_out = './test_pw.out'
+        path_to_espresso = '/usr/bin/pw.x'
+        if mpi:
+            command = 'mpirun -np '+str(mpi_Nprocessors)+' '+path_to_espresso+' < '+name_in +' > '+name_out
+        else:
+            command = path_to_espresso+' < '+name_in +' > '+name_out
+
+       # command = '/usr/bin/pw.x < '+name_in+' > '+name_out
+        print('qe wrapper attempting to run: '+command)
+        subprocess.Popen(command, shell=True, stdout=subprocess.PIPE).stdout.read()
```

Comment:
I used a try/except instead of subprocess.check_call as the latter would take twice as long (in the case of success) while giving same error info in the case of failure